data-structures/trie/trie.py

```diff
@@ -0,0 +1,40 @@
+class trienode:
+	def __init__(self,character):
+		self.character = character
+		self.nextnodes = {}
+		self.isEnd = False
+		self.dataNode = None
+	
+class trie: # use case: username/password
+	def __init__(self):
+		self.start = trienode('')
+	def insert(self,id,data): # returns true on successful insertion (data == password)
+		temp = self.start
+		for x in id:
+			temp.nextnodes[x] = temp.nextnodes.get(x,trienode(x))
+			temp = temp.nextnodes[x]
+		if temp.isEnd == True:
+			print("ID already Exists!!")
+			return False
+		temp.isEnd =True
+		temp.dataNode = data
+		return True
+	def findNode(self,id): # returns false if node doesn't exist and true, node if it does
+		temp = self.start
+		for x in id:
+			next = temp.nextnodes.get(x,None)
+			if next is None:
+				return False,None
+			temp = next
+		if temp.isEnd == True:
+			return True,temp
+		else:
+			return False,None
+
+if __name__ == '__main__':
+	t = trie()
+	t.insert('test1',"dummy1")
+	t.insert('test2',"dummy2")
+	print(t.findNode('test')) # (false,None)
+	a,b = t.findNode('test1')
+	print(a,b) # (true,memref)
```

Comment:
Instead of printing a `memref` use an appropriate message.
Refer to - https://www.geeksforgeeks.org/trie-insert-and-search/

Reply:
Done